### PR TITLE
Feature/tr 948/upgrade lti1.3 core from 3 to 6

### PR DESCRIPTION
**Pull request by alexeyvy**
Upgraded dependency on https://github.com/oat-sa/lib-lti1p3-core from major version 3 to 6.

How to test:
1. Set `"oat-sa/extension-tao-lti": "dev-feature/TR-948/upgrade-lti1.3-core-from-3-to-6"` as a dependency in your project's composer.json (just switching to the branch in the taoLti folder isn't enough, as we need composer to rebuild dependencies to have the newest `lib-lti1p3-core`.
2. in .env, set `FEATURE_FLAG_LTI1P3=true` to activate LTI 1.3 for CGEN.
3. Go to the back-office and create or update an LTI provider record (some fields are only visible after setting `FEATURE_FLAG_LTI1P3=true` in step 2, select 'LTI 1.3' as provider LTI version in the select box.
4. Go to Back-office > Preferences > LTI Platform key chain and validate this page behaves well (was broken when migrated to the new LTI core without the changes from this PR applied).
5. Create a test and pass it as a test taker through https://ltiapps.net/test/tc.php (specify LTI 1.3 as LTI version in the menu)
6. Other ideas of how to test it are welcome

First we will need to merge this PR https://github.com/oat-sa/lib-lti1p3-core/pull/104 which allows Guzzle:6 in lib-lti1p3-core to make it consistent with TAO CGEN. Once merged & released, I will replace this temporary reference in composer.json:
`"oat-sa/lib-lti1p3-core": "dev-fix/TR-948/downgrade-guzzle"`

with this:
"oat-sa/lib-lti1p3-core": "^6.0"

### CODE REVIEW

File: models/classes/Platform/Repository/Lti1p3RegistrationRepository.php [hunk 1]
```diff
@@ -100,8 +101,8 @@ private function translateKeyChain(TaoKeyChain $keyChain): KeyChain
         return new KeyChain(
             $keyChain->getIdentifier(),
             $keyChain->getName(),
-            $keyChain->getPublicKey()->getValue(),
-            $keyChain->getPrivateKey()->getValue()
+            new Key($keyChain->getPublicKey()->getValue()),
+            new Key($keyChain->getPrivateKey()->getValue()),
```
Comment (boajer) [inline models/classes/Platform/Repository/Lti1p3RegistrationRepository.php:105]:
```suggestion
            new Key($keyChain->getPrivateKey()->getValue())
```

**Comment by ekkinox** [comment]
> 6. Other ideas of how to test it are welcome

You can run locally the [LTI demo application](https://github.com/oat-sa/demo-lti1p3) (all is in readme how to run it), cause i'm not sure https://ltiapps.net/test/tc.php is a good way to test LTI 1.3 interactions

**Comment by ekkinox** [comment]
> First we will need to merge this PR [oat-sa/lib-lti1p3-core#104](https://github.com/oat-sa/lib-lti1p3-core/pull/104) which allows Guzzle:6 in lib-lti1p3-core to make it consistent with TAO CGEN. 

It's merged and released: [v6.0.1](https://github.com/oat-sa/lib-lti1p3-core/releases/tag/6.0.1) is nor available.
Thx for contributing !

**Review by m-lagarde — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Comment by emgolubev** [comment]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by emgolubev — APPROVED** [review]

**Review by boajer — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**